File: src/main/java/com/rwbymod/client/ColorizerDust.java
package com.rwbymod.client;

import com.rwbymod.misc.EnumDustType;

import net.minecraft.block.state.IBlockState;
import net.minecraft.client.renderer.color.IBlockColor;
import net.minecraft.client.renderer.color.IItemColor;
import net.minecraft.item.ItemStack;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.IBlockAccess;

public class ColorizerDust implements IItemColor, IBlockColor {
	private int tintIndex;
	
	public ColorizerDust() { this(0); }
	public ColorizerDust(int tintIndex) { this.tintIndex = tintIndex; }
	
	public int getColorFromItemstack(ItemStack stack, int tintIndex) {
		if(tintIndex != this.tintIndex || stack.getItemDamage() >= EnumDustType.values().length) return 0xFFFFFF;
        return EnumDustType.values()[stack.getItemDamage()].color;
	}
	
	public int colorMultiplier(IBlockState state, IBlockAccess worldIn, BlockPos pos, int tintIndex) {
		if(tintIndex != this.tintIndex || state.getBlock().getMetaFromState(state) >= EnumDustType.values().length) return 0x000000;
        return EnumDustType.values()[state.getBlock().getMetaFromState(state)].color;
	}
}

File: src/main/scala/com/rwbymod/dust/ColorizerDust.scala
package com.rwbymod.dust

import net.minecraft.client.renderer.color.IItemColor
import net.minecraft.client.renderer.color.IBlockColor
import net.minecraft.item.ItemStack
import net.minecraft.util.math.BlockPos
import net.minecraft.world.IBlockAccess
import net.minecraft.block.state.IBlockState

class ColorizerDust(private val tintIndex:Int=0) extends IItemColor with IBlockColor {

  def getColor(tintIndex:Int, meta:Int):Int = tintIndex match {
    case this.tintIndex => EnumDustType.values().lift(meta) match {
      case Some(i) => i.color
      case _ => 0xFFFFFF
    }
    case _ => 0xFFFFFF
  }

	def getColorFromItemstack(stack:ItemStack, ti:Int):Int = getColor(ti, stack.getItemDamage)
	def colorMultiplier(state:IBlockState, world:IBlockAccess, pos:BlockPos, ti:Int):Int = getColor(ti, state.getBlock.getMetaFromState(state))

}